src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/CrossChainTransferStore.cs

```diff
@@ -667,27 +655,37 @@ private async Task OnInsertBlocksAsync(DBreeze.Transactions.Transaction dbreezeT
             // Find transfer transactions in blocks
             foreach (Block block in blocks)
             {
-                IReadOnlyList<IDeposit> deposits = this.depositExtractor.ExtractDepositsFromBlock(block, blockHeight);
+                uint256 blockHash = block.GetHash();
```

Comment:
Then here I don't understand why you didn't use the existing ExtractWithdrawalsFromBlock(Block block, int blockHeight)?

Reply:
It is has an unwanted side-effect as discussed.